src/calendar/types.ts

```diff
@@ -1,21 +1,61 @@
+import { ActionComponent, DateInputVariant } from "../date-input";
+import { ActionType } from "../date-input/date-input-reducer";
+
 export interface CalendarProps {
+    // DateInput component props
+    /** Restrict selection to within this date range,
+     * in `YYYY-MM-DD` format. Example: `["2023-03-15", "2023-04-19"]` */
+    between?: [string, string] | undefined;
     /** Dates to disable in `YYYY-MM-DD` format. Example: `["2023-04-30"]` */
     disabledDates?: string[] | undefined;
     /** Called when date is selected, returns value in `YYYY-MM-DD` format or `"Invalid Date"` */
     onSelect?: ((value: string) => void) | undefined;
-    /** Selected date in `YYYY-MM-DD` format */
+    /** Called when day cell is hover, returns value in `YYYY-MM-DD` */
+    onHover?: ((value: string) => void) | undefined;
+    /** Selected start date in `YYYY-MM-DD` format */
     value?: string | undefined;
     /** The display type of the component. Values `standalone` | `input` */
     type?: CalendarType | undefined;
 
+    // DateInput with calendar component props
```

Comment:
the following props all seem to be for internal use with the date input only

we can introduce a standalone Calendar that exposes only these props
- between
- disabledDates
- onSelect
- onHover
- value

and internally uses this shared calendar component as `type=standalone`

and in the future we could have a RangedCalendar that also takes in
- endValue
- onSelect (with start + end)

and it would manage the `currentFocus` on the shared calendar component